### PR TITLE
refactor: re-arrange e2e tests 

### PR DESCRIPTION
**Pull request by KeranYang**
### Changes

* Merge HTTP, nats, and jetstream e2e tests into one single suite - a builtin-source suite. This is because the 3 test suites finish quickly and hence can be merged to reduce no. of suites.
* Add missing headers for e2e go files.
* Remove some early-day e2e test cases because they are now covered by other tests. e.g. `TestConditionalForwarding` is removed. This is to reduce no. of test cases.
* Remove `w.StreamVertexPodlogs` in test cases. It's printing a lot of logs in CI, making the log hard to read. We can bring it back when we want to debug CI workflow again.
* Rename sdk test suite to map test suite, because it's now completely for map. SDK tests are distributed to feature suites.

### TODOs (I will work on them on a separate PR)
* Some test cases can be merged into a single pipeline, which helps reduce test run time.
* ISB service termination periodically time out. Need to fix it.

### CODE REVIEW

File: test/sideinputs-e2e/sideinput-e2e_sink_source_test.go [hunk 1]
```diff
@@ -29,6 +29,7 @@ import (
 	. "github.com/numaproj/numaflow/test/fixtures"
 )
 
+// FIXME - this test is not executed in CI. We should include source and sink side-input tests in the CI.
```
[the remark this answers is not in the document]
Comment (KeranYang) [reply test/sideinputs-e2e/sideinput-e2e_sink_source_test.go:32]: Sorry, I missed the log. It is executed. I will remove this comment.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -151,7 +151,7 @@ jobs:
       fail-fast: false
       matrix:
         driver: [jetstream]
-        case: [e2e, diamond-e2e, transformer-e2e, kafka-e2e, http-e2e, nats-e2e, jetstream-e2e, sdks-e2e, reduce-one-e2e, reduce-two-e2e, udsource-e2e, api-e2e, sideinputs-e2e, idle-source-e2e, monovertex-e2e]
+        case: [e2e, diamond-e2e, transformer-e2e, kafka-e2e, map-e2e, reduce-one-e2e, reduce-two-e2e, udsource-e2e, api-e2e, sideinputs-e2e, idle-source-e2e, monovertex-e2e, builtin-source-e2e]
```
[the remark this answers is not in the document]
Comment (KeranYang) [reply .github/workflows/ci.yaml:154]: `diamond-e2e` and `e2e` are both one of the test suites that take long time to finish.

```
e2e: SLOW
--- PASS: TestFunctionalSuite (506.44s)
    --- PASS: TestFunctionalSuite/TestConditionalForwarding (102.57s) -> removed
    --- PASS: TestFunctionalSuite/TestCreateSimplePipeline (76.06s)
    --- PASS: TestFunctionalSuite/TestDropOnFull (56.58s)
    --- PASS: TestFunctionalSuite/TestFallbackSink (64.44s)
    --- PASS: TestFunctionalSuite/TestUDFFiltering (72.38s)
    --- PASS: TestFunctionalSuite/TestWatermarkEnabled (100.30s)

diamond: SLOW
--- PASS: TestDiamondSuite (486.51s)
    --- PASS: TestDiamondSuite/TestCycleBackward (120.52s)
    --- PASS: TestDiamondSuite/TestCycleToSelf (119.47s)
    --- PASS: TestDiamondSuite/TestJoinOnMapPipeline (67.50s)
    --- PASS: TestDiamondSuite/TestJoinOnReducePipeline (74.53s)
    --- PASS: TestDiamondSuite/TestJoinOnSinkVertex (69.52s)
```

[the remark this answers is not in the document]
Comment (whynowy) [reply .github/workflows/ci.yaml:154]: We can check later to see why they are so slow.

[the remark this answers is not in the document]
Comment (KeranYang) [reply .github/workflows/ci.yaml:154]: yes, I will do that as part of https://github.com/numaproj/numaflow/issues/1963

[the remark this answers is not in the document]
Comment (KeranYang) [reply .github/workflows/ci.yaml:154]: @whynowy could you approve this one? This is just a start to improving the CI experience. Also, I am not sure why the checks still include the removed http-e2e, nats-e2e etc. which is blocking me from merging this PR. Could you help me understand what's missing? Thanks!

**Review by whynowy — APPROVED** [review]

**Pull request merged**